packages/flutter/lib/src/widgets/image.dart

```diff
@@ -1180,6 +1180,21 @@ class _ImageState extends State<Image> with WidgetsBindingObserver {
                 : null,
       ),
     );
+    if (widget.errorBuilder != null) {
+      // Add an ephemeral listener to suppress errors after disposal.
+      // An `Image` widget with an `errorBuilder` is expected to report no
+      // errors through `FlutterError` in any cases, but an image stream must be
+      // dispose when the widget is unmounted, which requires having no more
+      // callbacks. The ephemeral listener is used to suppress the error while
+      // allowing the disposal. For more details, see
+      // https://github.com/flutter/flutter/issues/97077 .
+      if (newStream.completer == null) {
+        newStream.setCompleter(_EmptyImageCompleter());
```

Comment:
Can you explain a bit why this is needed?

Reply:
The ephemeralListener is only applicable to image completers. However, a stream might not be resolved with a non-null completer (for various reasons; it's nullable anyway). This tricks is also used somewhere else, such as https://github.com/flutter/flutter/blob/34f8df3c84c3edf690b2993f160ff221b6a0b183/packages/flutter/lib/src/painting/image_provider.dart#L398 .